Use jvm-example for testing bin/plugin

nik9000 opened the pull request:
Related to #12651

nik9000 committed:
Test: Use jvm-example for testing bin/plugin

Related to #12651

nik9000 commented:
Ping @tlrx for review - I'll be adding another pr to handle the TODO.

rjernst commented on qa/vagrant/src/test/resources/packaging/scripts/25_tar_plugins.bats:
```diff
@@ -59,40 +48,36 @@ setup() {
-    assert_file_exist "/tmp/elasticsearch/config/shield/users_roles"
-    assert_file_not_exist "/tmp/elasticsearch/plugins/shield"
+    # TODO add bin the jvm-example
+    # assert_file_not_exist "/tmp/elasticsearch/bin/shield"
```
Remove the reference to shield?

s1monw replied:
++

rjernst commented on qa/vagrant/src/test/resources/packaging/scripts/25_tar_plugins.bats:
```diff
@@ -59,40 +48,36 @@ setup() {
-    assert_file_exist "/tmp/elasticsearch/plugins/shield"
+    # Checks that the plugin is correctly installed
+    # TODO add bin to jvm-example
+    # assert_file_exist "/tmp/elasticsearch/bin/shield"
```
remove these old shield references?

rjernst commented:
LGTM, I would just remove all the commented out file checks for shield.

s1monw commented on qa/vagrant/src/test/resources/packaging/scripts/25_tar_plugins.bats:
```diff
@@ -109,44 +94,40 @@ setup() {
-    assert_file_exist "/tmp/elasticsearch/config/shield/users_roles"
-    assert_file_exist "$TEMP_PLUGINS_DIR/shield"
+    # Checks that the plugin is correctly installed
+    # TODO add bin to jvm-example
```
same here

s1monw commented on qa/vagrant/src/test/resources/packaging/scripts/25_tar_plugins.bats:
```diff
@@ -166,44 +147,40 @@ setup() {
-    assert_file_exist "$TEMP_CONFIG_DIR/shield/users_roles"
-    assert_file_exist "/tmp/elasticsearch/plugins/shield"
+    # TODO add bin to jvm-example
+    # assert_file_exist "/tmp/elasticsearch/bin/shield"
```
and here

nik9000 replied:
I'll drop the shield references next. I have a patch for adding bin to
jvm-example and will use that to replace all those bin checks. I can do
that in this commit if I merge it first and rebase. I don't have a link
because I'm on mobile.
On Aug 14, 2015 2:41 PM, "Simon Willnauer" <notifications@github.com> wrote:

> In qa/vagrant/src/test/resources/packaging/scripts/25_tar_plugins.bats
> <https://github.com/elastic/elasticsearch/pull/12895#discussion_r37122263>
> :
>
> >      [ "$status" -eq 0 ]
> >
> >      # Checks that Shield is correctly installed
> > -    assert_file_exist "/tmp/elasticsearch/bin/shield"
> > -    assert_file_exist "/tmp/elasticsearch/bin/shield/esusers"
> > -    assert_file_exist "/tmp/elasticsearch/bin/shield/syskeygen"
> > -    assert_file_exist "$TEMP_CONFIG_DIR/shield"
> > -    assert_file_exist "$TEMP_CONFIG_DIR/shield/role_mapping.yml"
> > -    assert_file_exist "$TEMP_CONFIG_DIR/shield/roles.yml"
> > -    assert_file_exist "$TEMP_CONFIG_DIR/shield/users"
> > -    assert_file_exist "$TEMP_CONFIG_DIR/shield/users_roles"
> > -    assert_file_exist "/tmp/elasticsearch/plugins/shield"
> > +    # TODO add bin to jvm-example
> > +    # assert_file_exist "/tmp/elasticsearch/bin/shield"
>
> and here
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/elasticsearch/pull/12895/files#r37122263>.
>

nik9000 replied:
#12897

nik9000 replied:
I've merged #12897 so I'll rebase this and use it to remove the shield references now.

nik9000 committed:
Add tests for plugins with bin directory

Also removes all mention of shield:
```bash
$ find $BATS -type f -exec grep -Hi shield {} \;
$
```

nik9000 commented:
Rebased, added tests for plugins with bin directory, and removed all mention of shield. Sorry for the rebase - I had to do it to get access to a plugin with a bin directory.

tlrx commented:
The changes LGTM but bats tests are failing at various stages on my laptop. I guess this is due to recent changes and not this PR specifically.

nik9000 commented:
> The changes LGTM but bats tests are failing at various stages on my laptop. I guess this is due to recent changes and not this PR specifically.

Hmm - they work for me:
```
vagrant destory -f
mvn clean install -DskipTests
mvn -pl qa/vagrant -Dtests.vagrant verify
```
and it all passes. That first line isn't usually required but if things are flakey its best to just rebuild the VMs....

nik9000 commented:
Ok - this has enough LGTMs that I've rebased it and will merge it to master. I'll help @tlrx work through vagrant issues when our days overlap better.
